packages/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -151,7 +808,130 @@ export default class FulfillmentModuleService<
   ): Promise<
     FulfillmentTypes.ServiceZoneDTO[] | FulfillmentTypes.ServiceZoneDTO
   > {
-    return []
+    const updatedServiceZones = await this.updateServiceZones_(
+      data,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<
+      FulfillmentTypes.ServiceZoneDTO | FulfillmentTypes.ServiceZoneDTO[]
+    >(updatedServiceZones, {
+      populate: true,
+    })
+  }
+
+  protected async updateServiceZones_(
+    data:
+      | FulfillmentTypes.UpdateServiceZoneDTO[]
+      | FulfillmentTypes.UpdateServiceZoneDTO,
+    sharedContext?: Context
+  ): Promise<TServiceZoneEntity | TServiceZoneEntity[]> {
+    const data_ = Array.isArray(data) ? data : [data]
+
+    const geoZonesToCreate: FulfillmentTypes.CreateGeoZoneDTO[] = []
+    const serviceZoneGeoZonesMap = new Map<
+      string,
+      Map<string, FulfillmentTypes.CreateGeoZoneDTO | { id: string }>
+    >()
+
+    const existingGeoZones: TGeoZoneEntity[] = []
+    let existingGeoZonesMap = new Map<string, TGeoZoneEntity>()
+
+    const geoZoneIds: string[] = []
+
+    data_.forEach((serviceZone) => {
+      if ("geo_zones" in serviceZone && serviceZone.geo_zones) {
+        const geo_zones = serviceZone.geo_zones
+
+        geo_zones.forEach((geoZone) => {
+          if ("id" in geoZone) {
+            geoZoneIds.push(geoZone.id)
+          }
+        })
+      }
+    })
+
+    if (geoZoneIds.length) {
+      existingGeoZones.push(
+        ...(await this.geoZoneService_.list(
+          {
+            id: geoZoneIds,
+          },
+          {},
+          sharedContext
+        ))
+      )
```

Comment:
suggestion: In these case, I'd rather have a separate API call. Think it makes for cleaner code, even thought it increases the footprint

Reply:
🤣 you are right, i usually dont do that but i went crazy, i will store it separately before pushing